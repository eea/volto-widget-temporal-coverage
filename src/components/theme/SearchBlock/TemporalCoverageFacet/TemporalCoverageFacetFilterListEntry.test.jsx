import React from 'react';
import { render } from '@testing-library/react';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-intl-redux';
import thunk from 'redux-thunk';
import TemporalCoverageFacetFilterListEntry from './TemporalCoverageFacetFilterListEntry';

const mockStore = configureStore([thunk]);

it('renders an temporal widget edit component with existing temporal ranges and single values', () => {
  const store = mockStore({
    intl: {
      locale: 'en',
      messages: {},
    },
    temporal: [
      { value: '1890', label: '1890' },
      { value: '1900-1905', label: '1900-1905' },
      { value: '1915', label: '1915' },
    ],
  });

  const component = render(
    <Provider store={store}>
      <TemporalCoverageFacetFilterListEntry facets={[]} />
    </Provider>,
  );
  expect(component.container).toMatchSnapshot();
});
