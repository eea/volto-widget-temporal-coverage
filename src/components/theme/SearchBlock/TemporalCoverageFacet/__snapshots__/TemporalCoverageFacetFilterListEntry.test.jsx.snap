// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders an temporal widget edit component with existing temporal ranges and single values 1`] = `
<div>
  <div
    class="ui small label"
  >
    No
    <i
      aria-hidden="true"
      class="delete icon"
    />
  </div>
</div>
`;
